CONVENTIONS.md

```diff
@@ -0,0 +1,55 @@
+
+Likewise all methods that can accept a null value should be annotated using the JetBrains `@Nullable` right before the method parameter. For example, a method named `food` that can accept a null value should be written as follows `public Item.Settings food(@Nullable FoodComponent food)...`.
+
+## Flow Control
+
+Wherever it makes sense the ternary operator ought to be preferred over an if statement for the sake of brevity, but never at the expense of readability. If the statement contains a long if condition it should not be turned into a ternary operator because the line will be too long and the statement will lose readability. If the statement contains a short if else statement and can be expressed simply as a ternary operator, that ought to be preferred.
+
+## Enums
+
+Enum values that are referenced internally (values referenced within methods associated with their own enum definition) must always use `Enum.VALUE` never simply `VALUE`.
+
+## Mixins
+
+Regular mixin classes must be named the same name as the target class they are being mixed into, with "Mixin" appended to the end of the class name. As an example, a mixin class into the class `MinecraftServer` should be named `MinecraftServerMixin`.
+
+Likewise, accessor mixin classes must be named the same name as the target class they are being mixed into, with "Accessor" appended to the end of the class name. As an example, the accessor class into the class `MinecraftServer` should be named `MinecraftServerAccessor`.
+
+"Duck interfaces" (Interfaces that are implemented onto classes using mixins) should be named the same name as the target class that implements the interface, with a series of adjectives at the front of the class name describing the contents or the implementation use case of the interface, along with "Hooks" at the end of the class name. For example, a duck interface implemented by `PlayerEntity` through mixins in order to add a mana player attribute should be named `ManaPlayerEntityHooks`.
+
+Mixins that shadow final fields must use the `@Final` annotation provided by the mixin library.
+
+Mixin injection and redirect methods must be named based upon the specific function of the injection or redirection, not the method that they are injecting into or redirecting. For example a method inject that registers commands on the dedicated server ought to be named `registerCommands` not the name of the target method, in this case `setupServer`.
+
+If a shadowed method needs to be prefixed because of a conflict it must use the prefix `minecraft$` for methods in minecraft, `[namespace]$` for projects with a different namespace (which should always match the mod id), or if neither of those are available `[package name]$`. Take for example a situation where the shadowed method `fromJson` conflicts with a new `fromJson` method. The shadowed method should be renamed to `minecraft$fromJson` and annotated as follows `@Shadow(prefix = "minecraft$")`.
```

Comment:
This would occur very rarely, but I would lean towards the default prefix of `shadow$`.

That explicitly specifies that the method being shadowed in name. Sponge guys have had pretty good milage with that default prefix for everything, but fabric api would probably use it for conflicts only.